docs/sources/k6-studio/components/test-generator.md

```diff
@@ -0,0 +1,115 @@
+
+### Test data
+
+The test data option allows you to define variables, which you can then use in your [custom code rules](#custom-code-rule).
+
+After you define a variable, you can refer to them in your custom code rules by using: `VARS["VARIABLE_NAME"]`.
+
+## Allowed hosts
+
+The **Allowed hosts** option lets you configure which hosts you want to include in your test script. It shows all the hosts that are identified from your HAR recording file, and you can choose to select which ones you want to include or remove from your test script.
+
+For performance testing, it's common to not include static assets, or remove hosts from 3rd-party services from your tests, but you can include them depending on your use case.
+
+## Rules
+
+Test rules are rules you can add and configure to your test generator, that allow you to customize the generated test script. These rules can use the information from a test recording, to then generate code changes to help make your test scripts more reliable, and reusable.
+
+The available rules are:
+
+- Verification rule
+- Correlation rule
+- Custom code rule
+
+You can add multiple correlation and custom code rules to your test generator.
+
+### Verification rule
+
+The verification rule is a default rule that's created for every test generator. It adds a [check](https://grafana.com/docs/k6/latest/using-k6/checks/) statement after every request that validates if the response status code is the same as the one from the test recording.
+
+The verification rule is useful for making sure that your systems are performing as expected. Failed checks do not cause a test to abort, or interrupt a test run. Checks generate a metric that you can inspect after a test run is completed to understand how your system performed.
+
+For more details about checks, refer to [Checks](https://grafana.com/docs/k6/latest/using-k6/checks/).
+
+### Correlation rule
+
+The correlation rule allows you to extract data from your test recording, and reuse it across your test script. This is a common use case when working with applications that have unique IDs that are generated for a request, and then are passed to every subsequent request, for example.
+
+The correlation rule includes an Extractor and a Replacer. They both have the same configuration options, but the Extractor can be used to find and extract the value you want to use across the script, and the Replacer can be used to find the places where you want to replace and use the extracted value.
+
+The configuration fields are:
+
+- **Filter**: Define a request path that this filter applies to. Plain text and regular expression are supported.
+- **Target**: Select the headers, body, or URL path as the target for the extractor or replacer.
+- **Type**: Select Begin-End or Regex as the way to search for the value to be extracted or replaced.
```

Comment:
we also support `json` as type
It allows you to define a json key to retrieve the value from the body. 
ex.
```
title
[0].title
```